### PR TITLE
skip the validation for external projects

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/autorest.csharp/issues/3179

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs [hunk 1]
```diff
@@ -95,7 +95,10 @@ public enum UnreferencedTypesHandlingOption
                 _absoluteProjectFolder = Path.GetFullPath(Path.Combine(outputFolder, projectFolder));
             }
 
-            if (publicClients && generation1ConvenienceClient)
+            var isAzureProject = ns != null && (ns.StartsWith("Azure.") || ns.StartsWith("Microsoft.Azure"));
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs:98]: This check skips the validation when the namespace is not set. Should we do this?

Comment (m-nash) [reply src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs:98]: In our build context we assign library name to the namespace if its not set don't we and library name has to always be set?

If that is the case perhaps we promote that default logic into configuration and remove it from buildcontext.

**Comment by ArcturusZhang** [comment]
@m-nash I made a few changes, could you take a look again?

**Comment by ArcturusZhang** [comment]
@m-nash latest regen is here: https://github.com/Azure/azure-sdk-for-net/pull/35727

**Review by m-nash — APPROVED** [review]

**Pull request merged**